namecheap/resource_namecheap_record.go

```diff
@@ -79,14 +83,33 @@ func resourceNameCheapRecordCreate(d *schema.ResourceData, meta interface{}) err
 		TTL:        d.Get("ttl").(int),
 	}
 
-	_, err := client.AddRecord(d.Get("domain").(string), &record)
-
-	if err != nil {
-		mutex.Unlock()
-		return fmt.Errorf("Failed to create namecheap Record: %s", err)
+	apiThrottleBackoffTime := 2
+
+	for {
+		_, err := client.AddRecord(d.Get("domain").(string), &record)
+		if err != nil {
+			log.Printf("[TRACE] Err: %v", err.Error())
+
+			if strings.Contains(err.Error(), "expected element type <ApiResponse> but have <html>") {
+				log.Printf("[WARN] Bad Namecheap API response received, backing off for %d seconds...",
+					apiThrottleBackoffTime)
+				time.Sleep(time.Duration(apiThrottleBackoffTime) * time.Second)
+				apiThrottleBackoffTime = apiThrottleBackoffTime * ncBackoffMultiplier
+				if apiThrottleBackoffTime > ncMaxThrottleRetry {
+					log.Printf("[ERROR] API Retry Limit Reached. Couldn't find namecheap record: %v", err)
+					break
```

Comment:
I left the mutex.Unlock at the end of the method and just removed these in the middle. These must be unnecessary since we are unlocking at the end anyway.

Reply:
Yea. We need to just unlock before returning or handing off control flow to another func that expects to lock.